clientupdate/clientupdate.go

```diff
@@ -289,6 +289,7 @@ func (up *Updater) updateSynology() error {
 	if err != nil {
 		return err
 	}
+	defer os.RemoveAll(spkDir)
```

Comment:
we might want to do something like `cleanupOldMSIs` here as well? if we do an update from tailscaled, it will restart on a successful update and this deferred function won't run

Reply:
Good idea, I generalized `cleanupOldMSIs`